### PR TITLE
Fix Docker setup to ease entry for new contributors

### PR DESCRIPTION
**Pull request by pacodelaluna**
#### What? Why?
Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/8823.

#### What should we test?
Test locally the Docker setup running `docker-compose up --build`, and then `docker exec -it openfoodnetwork_web_1 bundle exec rake db:schema:load db:seed ofn:sample_data` to seed the database.

#### Release notes
- Fix and update Docker configuration

Changelog Category: Technical changes

#### Documentation updates
- Update the Docker README file
- Update the repository wiki

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -41,13 +43,13 @@ RUN git clone --depth 1 https://github.com/rbenv/rbenv.git ${RBENV_ROOT} && \
     echo 'eval "$(rbenv init -)"' >> /etc/profile.d/rbenv.sh && \
     RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install $(cat .ruby-version) && \
     rbenv global $(cat .ruby-version) && \
-    gem install bundler --version=1.17.3
+    gem install bundler --version=2.3.5
```
Comment (mkllnk) [inline Dockerfile:46]: We actually use version 2.1.4. You can use `./script/install-bundler` for this.

File: docker-compose.yml [hunk 2]
```diff
@@ -36,7 +36,8 @@ services:
     command: >
       bash -c "wait-for-it -t 30 db:5432 &&
                rm -f tmp/pids/server.pid &&
-               (bundle check || bundle install)
+               (bundle check || bundle install) &&
```
Comment (mkllnk) [inline docker-compose.yml:39]: I wonder why `bundle check` was in here. I think that `bundle install` is checking that already.

**Review by mkllnk — APPROVED** [review]
Thanks!

File: docker-compose.yml (hunk 2 above)
Comment (pacodelaluna) [reply docker-compose.yml:39]: It is a habit I got from previous projects, it is a bit more verbose, listing first the missing dependencies and installing them after. We can just keep `bundle install` if you prefer.

**Comment by pacodelaluna** [comment]
This Docker configuration is now working. We can improve it later with specific OFN Docker image later maybe, we need to check if we really need it.

I am removing the Draft flag, this PR can be tested locally. For testers, don't hesitate to ping me on Slack if needed.

**Review by mkllnk — APPROVED** [review]

**Comment by RachL** [comment]
@openfoodfoundation/core-devs shall we merge?

**Comment by jibees** [comment]
> @openfoodfoundation/core-devs shall we merge?

Moving to `Dev test ready` ?

**Review by jibees — APPROVED** [review]

**Comment by RachL** [comment]
@jibees not sure the testing team can test this? Dev test ready?

**Review by georgethoppil — APPROVED** [review]

**Comment by mkllnk** [comment]
@georgethoppil Did you test this locally?

**Comment by georgethoppil** [comment]
@mkllnk yeah it worked for me. There was a rake abort on seeding tho but I believe there was a ticket for it if I am not mistaken


```
[ofn:sample_data] Maryse's Private Shop permits everybody to do everything.
[ofn:sample_data] Creating fees:
[ofn:sample_data] - Penny's Profile charges markup
rake aborted!
NoMethodError: undefined method `bind' for nil:NilClass
/usr/src/app/lib/spree/localized_number.rb:31:in `block (2 levels) in localize_number'
/bundles/ruby/3.0.0/gems/activemodel-6.1.4.4/lib/active_model/attribute_assignment.rb:49:in `public_send'
/bundles/ruby/3.0.0/gems/activemodel-6.1.4.4/lib/active_model/attribute_assignment.rb:49:in `_assign_attribute'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/attribute_assignment.rb:21:in `block in _assign_attributes'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/attribute_assignment.rb:13:in `each'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/attribute_assignment.rb:13:in `_assign_attributes'
/bundles/ruby/3.0.0/gems/activemodel-6.1.4.4/lib/active_model/attribute_assignment.rb:34:in `assign_attributes'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/core.rb:510:in `initialize'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/inheritance.rb:72:in `new'
/bundles/ruby/3.0.0/gems/activerecord-6.1.4.4/lib/active_record/inheritance.rb:72:in `new'
/usr/src/app/lib/tasks/sample_data/fee_factory.rb:15:in `block in create_samples'
/usr/src/app/lib/tasks/sample_data/fee_factory.rb:11:in `each'
/usr/src/app/lib/tasks/sample_data/fee_factory.rb:11:in `create_samples'
/usr/src/app/lib/tasks/sample_data.rake:38:in `block (2 levels) in <main>'
/bundles/ruby/3.0.0/gems/bugsnag-6.24.1/lib/bugsnag/integrations/rake.rb:20:in `execute'
/bundles/ruby/3.0.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/usr/local/src/rbenv/versions/3.0.3/bin/bundle:23:in `load'
/usr/local/src/rbenv/versions/3.0.3/bin/bundle:23:in `<main>'
Tasks: TOP => ofn:sample_data
(See full trace by running task with --trace)
```

**Comment by mkllnk** [comment]
That's a different error. It may be that the locale doesn't match the system's locale. Anyway. this is an improvement and I'll merge it.

**Pull request merged**